### PR TITLE
Add network policies for rhoai

### PR DESCRIPTION
**Pull request by rajivnathan**
e2e tests: https://github.com/codeready-toolchain/toolchain-e2e/pull/960

### CODE REVIEW

**Review by VedantMahabaleshwarkar — APPROVED** [review]
lgtm

**Review by alexeykazakov — APPROVED** [review]
Do we want to add it to the base tier too? We can do it in a separate PR though.

**Comment by rajivnathan** [comment]
>Do we want to add it to the base tier too? We can do it in a separate PR though.
Added the network policies to base tier too since the tests are expecting them

**Comment by alexeykazakov** [comment]
```
"unable to create resource of kind: NetworkPolicy, version: v1: unable to create resource of kind: NetworkPolicy, version: v1: un"...,
```

File: deploy/templates/nstemplatetiers/base/ns_stage.yaml [hunk 1]
```diff
@@ -188,6 +188,42 @@ objects:
     podSelector: {}
     policyTypes:
     - Ingress
+- apiVersion: networking.k8s.io/v1
+  kind: NetworkPolicy
+  metadata:
+    name: allow-from-redhat-ods-app-to-mariadb
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/base/ns_stage.yaml:194]:
```suggestion
    name: allow-from-redhat-ods-app-to-mariadb
    namespace: ${SPACE_NAME}-stage
```

File: deploy/templates/nstemplatetiers/base/ns_stage.yaml [hunk 2]
```diff
@@ -188,6 +188,42 @@ objects:
...
+                app.kubernetes.io/name: data-science-pipelines-operator
+            namespaceSelector:
+              matchLabels:
+                kubernetes.io/metadata.name: redhat-ods-applications
+    policyTypes:
+      - Ingress
+- apiVersion: networking.k8s.io/v1
+  kind: NetworkPolicy
+  metadata:
+    name: allow-from-redhat-ods-app-to-mm
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/base/ns_stage.yaml:215]:
```suggestion
    name: allow-from-redhat-ods-app-to-mm
    namespace: ${SPACE_NAME}-stage
```

File: deploy/templates/nstemplatetiers/base1ns/ns_dev.yaml [hunk 3]
```diff
@@ -214,6 +214,42 @@ objects:
     podSelector: {}
     policyTypes:
     - Ingress
+- apiVersion: networking.k8s.io/v1
+  kind: NetworkPolicy
+  metadata:
+    name: allow-from-redhat-ods-app-to-mariadb
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/base1ns/ns_dev.yaml:220]:
```suggestion
    name: allow-from-redhat-ods-app-to-mariadb
    namespace: ${SPACE_NAME}-dev
```

File: deploy/templates/nstemplatetiers/base1ns/ns_dev.yaml [hunk 4]
```diff
@@ -214,6 +214,42 @@ objects:
...
+                app.kubernetes.io/name: data-science-pipelines-operator
+            namespaceSelector:
+              matchLabels:
+                kubernetes.io/metadata.name: redhat-ods-applications
+    policyTypes:
+      - Ingress
+- apiVersion: networking.k8s.io/v1
+  kind: NetworkPolicy
+  metadata:
+    name: allow-from-redhat-ods-app-to-mm
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/base1ns/ns_dev.yaml:241]:
```suggestion
    name: allow-from-redhat-ods-app-to-mm
    namespace: ${SPACE_NAME}-dev
```

File: deploy/templates/nstemplatetiers/base/ns_dev.yaml [hunk 5]
```diff
@@ -188,6 +188,42 @@ objects:
     podSelector: {}
     policyTypes:
     - Ingress
+- apiVersion: networking.k8s.io/v1
+  kind: NetworkPolicy
+  metadata:
+    name: allow-from-redhat-ods-app-to-mariadb
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/base/ns_dev.yaml:194]:
```suggestion
    name: allow-from-redhat-ods-app-to-mariadb
    namespace: ${SPACE_NAME}-dev
```

File: deploy/templates/nstemplatetiers/base/ns_dev.yaml [hunk 6]
```diff
@@ -188,6 +188,42 @@ objects:
...
+                app.kubernetes.io/name: data-science-pipelines-operator
+            namespaceSelector:
+              matchLabels:
+                kubernetes.io/metadata.name: redhat-ods-applications
+    policyTypes:
+      - Ingress
+- apiVersion: networking.k8s.io/v1
+  kind: NetworkPolicy
+  metadata:
+    name: allow-from-redhat-ods-app-to-mm
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/base/ns_dev.yaml:215]:
```suggestion
    name: allow-from-redhat-ods-app-to-mm
    namespace: ${SPACE_NAME}-dev
```

**Comment by alexeykazakov** [comment]
```
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -9,6 +9,21 @@
        	            	   (v1.NetworkPolicyIngressRule) {
        	            	-   Ports: ([]v1.NetworkPolicyPort) <nil>,
        	            	+   Ports: ([]v1.NetworkPolicyPort) (len=1) {
        	            	+    (v1.NetworkPolicyPort) {
        	            	+     Protocol: (*v1.Protocol)((len=3) "TCP"),
        	            	+     Port: (*intstr.IntOrString)({
        	            	+      Type: (intstr.Type) 0,
        	            	+      IntVal: (int32) 3306,
        	            	+      StrVal: (string) ""
        	            	+     }),
        	            	+     EndPort: (*int32)(<nil>)
        	            	+    }
        	            	+   },
        	            	    From: ([]v1.NetworkPolicyPeer) (len=1) {
        	            	     (v1.NetworkPolicyPeer) {
        	            	-     PodSelector: (*v1.LabelSelector)(<nil>),
        	            	+     PodSelector: (*v1.LabelSelector)({
        	            	+      MatchLabels: (map[string]string) (len=1) {
        	            	+       (string) (len=22) "app.kubernetes.io/name": (string) (len=31) "data-science-pipelines-operator"
        	            	+      },
        	            	+      MatchExpressions: ([]v1.LabelSelectorRequirement) <nil>
        	            	+     }),
        	            	      NamespaceSelector: (*v1.LabelSelector)({
```

**Pull request merged**